organization/src/main/scala/com/improving/app/organization/OrganizationServiceImpl.scala

```diff
@@ -0,0 +1,78 @@
+import com.improving.app.organization.domain.{ActivateOrganization, EstablishOrganization, GetOrganizationInfo, Organization, OrganizationActivated, OrganizationEstablished, OrganizationInfo, OrganizationSuspended, OrganizationTerminated, SuspendOrganization, TerminateOrganization}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+class OrganizationServiceImpl(sys: ActorSystem[_]) extends OrganizationService {
+  private implicit val system: ActorSystem[_] = sys
+  implicit val timeout: Timeout = 5.minute
+  implicit val executor = system.executionContext
+
+  val sharding = ClusterSharding(system)
+
+  sharding.init(Entity(Organization.TypeKey)(
+    createBehavior = entityContext =>
+      Organization(
+        PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId)
+      )
+  ))
+
+  Cluster(system).manager ! Join(Cluster(system).selfMember.address)
+
+  private def exceptionHandler(exception: Throwable): GrpcServiceException = {
+    GrpcServiceException(
+      code = Code.INVALID_ARGUMENT,
+      message = exception.getMessage,
+      details = Seq(new LocalizedMessage("EN", exception.getMessage))
+    )
+  }
+
+  override def establishOrganization(in: EstablishOrganization): Future[OrganizationEstablished] = {
+    val result = sharding.entityRefFor(Organization.TypeKey, in.organizationId.get.id)
+      .ask(ref => Organization.OrganizationRequestEnvelope(in, ref))
+    result.transform(
+      result => result.getValue.asMessage.sealedValue.organizationEstablished.get,
+      exception => exceptionHandler(exception)
+    )
+  }
+
+  override def activateOrganization(in: ActivateOrganization): Future[OrganizationActivated] = {
+    val result = sharding.entityRefFor(Organization.TypeKey, in.organizationId.get.id)
+      .ask(ref => Organization.OrganizationRequestEnvelope(in, ref))
+    result.transform(
+      result => result.getValue.asMessage.sealedValue.organizationActivated.get,
+      exception => exceptionHandler(exception)
+    )
+  }
+
+  override def suspendOrganization(in: SuspendOrganization): Future[OrganizationSuspended] = {
+    val result = sharding.entityRefFor(Organization.TypeKey, in.organizationId.get.id)
+      .ask(ref => Organization.OrganizationRequestEnvelope(in, ref))
+    result.transform(
+      result => result.getValue.asMessage.sealedValue.organizationSuspended.get,
+      exception => exceptionHandler(exception)
+    )
+  }
+
+  override def terminateOrganization(in: TerminateOrganization): Future[OrganizationTerminated] = {
+    val result = sharding.entityRefFor(Organization.TypeKey, in.organizationId.get.id)
+      .ask(ref => Organization.OrganizationRequestEnvelope(in, ref))
+    result.transform(
+      result => result.getValue.asMessage.sealedValue.organizationTerminated.get,
+      exception => exceptionHandler(exception)
+    )
+  }
+
+  override def getOrganizationInfo(in: GetOrganizationInfo): Future[OrganizationInfo] = ???
```

Comment:
Is this intended? If not, this also means your spec files doesn't have sufficient coverage

Reply:
There's another ticket for this: IA-162. I wanted to get this query into the initial protocol buffer implementation to get the types settled, but I also wanted to keep the change set small to avoid getting bogged down

Reply:
Understood with regards to having this done in a different ticket. However, what does IA-162 have to do with getOrganizationInfo? Isn't IA-162 just consolidating together validation code?

Reply:
oops, IA-163 is the correct ticket

Reply:
sounds good